toolbox/util/githubClient.go

```diff
@@ -284,26 +284,69 @@ func (g GithubClient) GetTagCommitSHA(repo, tag string) (string, error) {
 }
 
 // GetCommitCreationTime gets the time when the commit identified by sha is created
-func (g GithubClient) GetCommitCreationTime(repo, sha string) (*time.Time, error) {
+func (g GithubClient) GetCommitCreationTime(repo, sha string) (time.Time, error) {
 	commit, _, err := g.client.Git.GetCommit(
 		context.Background(), g.owner, repo, sha)
 	if err != nil {
 		log.Printf("Failed to get commit %s", sha)
-		return nil, err
+		return time.Time{}, err
 	}
-	return (*(*commit).Author).Date, nil
+	return commit.Author.GetDate(), nil
 }
 
 // GetCommitCreationTimeByTag finds the time when the commit pointed by a tag is created
 // Note that SHA of the tag is different from the commit SHA
-func (g GithubClient) GetCommitCreationTimeByTag(repo, tag string) (*time.Time, error) {
+func (g GithubClient) GetCommitCreationTimeByTag(repo, tag string) (time.Time, error) {
 	commitSHA, err := g.GetTagCommitSHA(repo, tag)
 	if err != nil {
-		return nil, err
+		return time.Time{}, err
 	}
 	return g.GetCommitCreationTime(repo, commitSHA)
 }
 
+// GetReleaseCreationTime gets the creation time of a release tag (0.1.6, 0.2.7)
+func (g GithubClient) GetReleaseCreationTime(repo, tag string) (createTime time.Time, err error) {
```

Comment:
You put `(createTime time.Time, err error) ` as return types. Does that declare those vars implicitly for you? cause I also notice createTime and err are assigned values directly as in `createTime, err = g.getReleaseTagCreationTime(repo, tag)`

Reply:
Yes, we can do that.